official/keras_application_models/benchmark_main.py

```diff
@@ -0,0 +1,242 @@
+    "densenet121": tf.keras.applications.DenseNet121,
+    "densenet169": tf.keras.applications.DenseNet169,
+    "densenet201": tf.keras.applications.DenseNet201,
+    "nasnetlarge": tf.keras.applications.NASNetLarge,
+    "nasnetmobile": tf.keras.applications.NASNetMobile,
+}
+
+
+def get_model_callbacks(benchmark_level, callbacks_list, batch_size=None,
+                        epoch_size=None, metrics=None, metric_logger=None):
+  """Create a list of callbacks.
+
+  Args:
+    benchmark_level: a string to indicate if the benchmark is `batch_based` or
+      `epoch_based`. Check `model_callbacks.py` for more details.
+    callbacks_list: a list of strings to name desired callbacks. Allowed:
+      ExamplesPerSecondCallback, LoggingMetricCallback, which are defined
+      in CALLBACKS.
+    batch_size: an int of batch size.
+    epoch_size: an int of epoch size, which is the total number of training
+      samples in one epoch.
+    metrics: `dict` that maps string-valued tags to metric names, or `iterable`
+      of metric names.
+    metric_logger: instance of `BenchmarkLogger`, the benchmark logger that
+      callback should use to write the log.
+
+  Returns:
+    callbacks: a list of callbacks.
+
+  Raises:
+    ValueError: if an unrecognized callback name is passed.
+  """
+  if not callbacks_list:
+    return []
+
+  batch_based = True if benchmark_level == "batch_based" else None
+  epoch_based = True if benchmark_level == "epoch_based" else None
+
+  callbacks = []
+  for callback in callbacks_list:
+    callback_name = callback.strip().lower()
+    if callback_name not in model_callbacks.CALLBACKS:
+      raise ValueError("Unrecognized callback requested: {}".format(callback))
+
+    if callback_name == model_callbacks.CALLBACKS[0]:
+      exp_per_second_callback = model_callbacks.ExamplesPerSecondCallback(
+          batch_size=batch_size, epoch_size=epoch_size,
+          batch_based=batch_based, epoch_based=epoch_based,
+          metric_logger=metric_logger)
+      callbacks.append(exp_per_second_callback)
+    else:
+      logging_metric_callback = model_callbacks.LoggingMetricCallback(
+          metrics=metrics, metric_logger=metric_logger,
+          batch_based=batch_based, epoch_based=epoch_based)
+      callbacks.append(logging_metric_callback)
+
+  return callbacks
+
+
+def generate_synthetic_input_dataset(image_size, batch_size):
+  """Generate synthetic dataset."""
+  input_shape = (batch_size,) + image_size + (_NUM_CHANNELS,)
+
+  images = tf.zeros(input_shape, dtype=tf.float32)
+  labels = tf.zeros((batch_size, _NUM_CLASSES), dtype=tf.float32)
+
+  return tf.data.Dataset.from_tensors((images, labels)).repeat()
+
+
+def get_default_image_size(model):
+  """Provide default image size for each model."""
+  image_size = (224, 224)
+  if model in ["inception", "xception", "inceptionresnet"]:
+    image_size = (299, 299)
+  elif model in ["nasnetlarge"]:
+    image_size = (331, 331)
+  return image_size
```

Comment:
This should only be necessary for the synth data. Can we make the function private, and name it something like _get_default_image_size_for_synthetic_data? If the user ever needs to do this under normal circumstances, then we should expose those numbers from the keras models themselves.

Reply:
These image sizes are the default values when feeding dataset to models, and are required if no other image sizes are provided. Of course users can also specify the image size. Should we give this option to users?